calliope/backend/helper_functions.py

```diff
@@ -81,22 +84,38 @@ def _squeeze_primary_carriers(component, carrier_tier):
     return _squeeze_primary_carriers
 
 
-def get_connected_link(model_data, **kwargs):
-    def _get_connected_link(component):
-        dims = [i for i in component.dims if i in ["techs", "nodes"]]
-        remote_nodes = model_data.link_remote_nodes.stack(idx=dims).dropna("idx")
-        remote_techs = model_data.link_remote_techs.stack(idx=dims).dropna("idx")
-        remote_component_items = component.sel(techs=remote_techs, nodes=remote_nodes)
+def select_from_lookup_table(model_data: xr.Dataset, **kwargs) -> Callable:
+    def _select_from_lookup_table(
+        component: xr.DataArray, **slice_dims
+    ) -> xr.DataArray:
+        dims = set(slice_dims.keys())
+        if dims.difference(component.dims):
+            raise BackendError(
+                f"Cannot select items from `{component.name}` on the dimensions {dims} since the array is not indexed over the dimensions {dims.difference(component.dims)}"
+            )
+        if any(
+            dim not in dim_slicer.dims
+            for dim in dims
+            for dim_slicer in slice_dims.values()
+        ):
+            raise BackendError(
+                f"All lookup tables used to select items from `{component.name}` must be indexed over the dimensions {dims}"
+            )
+
+        sliced_component = component.sel(
+            **{
+                dim_name: model_data[dim_slicer.name].stack(idx=dims).dropna("idx")
```

Comment:
What is happening here?

Reply:
Will give more descriptive name and an inline comment.